gcp/cloud_run_functions/run_qualtrics_scheduling/utils/validation_utils.py

```
@@ -172,3 +172,78 @@ def extract_participant_data(
+                payload.response_id,
+            )
+            return None
+
+        try:
+            selected_date = date.fromisoformat(payload.selected_date)
+        except (ValueError, AttributeError):
+            logger.error(
+                "Invalid date format for Connect response %s: '%s'",
+                payload.response_id,
+                payload.selected_date,
+            )
```

Ruff (0.15.15)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)